docs/coverage-reporter/index.md

```diff
@@ -0,0 +1,444 @@
+
+    ```bash
+    export CODACY_API_BASE_URL=<your Codacy instance URL>
+    ```
+
+1.  Run Codacy Coverage Reporter **on the root of the locally checked out branch of your Git repository**, specifying the relative path to the coverage report to upload:
+
+    ```bash
+    bash <(curl -Ls https://coverage.codacy.com/get.sh) report -r <coverage report file name>
+    ```
+
+    Check the console output to validate that the Codacy Coverage Reporter **detected the correct commit SHA-1 hash** and **successfully uploaded** the coverage data to Codacy. If you need help, [check the troubleshooting page](troubleshooting-coverage-cli-issues.md) for solutions to the most common issues while running the CLI.
+
+    !!! note
+        Be sure to also check the [instructions for more advanced scenarios](uploading-coverage-in-advanced-scenarios.md) while uploading the coverage data to Codacy, such as when running parallel tests, using monorepos, or testing source code in multiple or unsupported languages.
+
+## 3. Validating that the coverage setup is complete {: id="validating-coverage"}
+
+Codacy displays the code coverage in each branch, as well as the evolution of code coverage between commits and the code coverage variation introduced by pull requests.
+
+Because of this, to ensure that all code coverage metrics are available on Codacy, you must have successfully uploaded coverage data and analyzed:
+
+-   The last two commits in each branch
+-   The common ancestor commit of each pull request branch and its target branch
+
+!!! note "Example"
+    The example below shows that after pushing a commit that correctly sets up coverage on the default branch:
+
+    -   Codacy will report coverage metrics for all subsequent commits and pull requests relative to the default branch.
+
+    -   Codacy won't report coverage metrics for commits and pull requests that are relative to older branches where the coverage setup wasn't performed yet.
+
+        To solve this issue, you can rebase the old feature branch to update the common ancestor commit to one that already has coverage data.
+
+    ![Setting up coverage on the default branch](images/coverage-validate.png)
+
+Follow these instructions to validate that your coverage setup is working correctly:
+
+1.  On Codacy, open your **Repository Settings**, tab **Coverage**, and observe the list of the most recent 50 coverage reports in the section **Test your integration**.
+
+    Make sure that Codacy receives and processes the coverage data successfully for **at least two commits**.
+
+    ![Testing the coverage integration](images/coverage-test-integration.png)
+
+    If there are commits with a status different from **Processed**, please follow the troubleshooting instructions for the corresponding error status and click the button **Test integration** to display any new coverage reports uploaded to Codacy.
+
+    ### Commit not found {: id="status-commit-not-found" style="color: #EF5454;"}
```

Comment:
Codacy complains about these indented headings. Thoughts on solving/ignoring this?

Reply:
Yes, I knew I was hammering the Markdown syntax a bit when I created these section titles **inside** a list item. :sweat_smile: 

For this exceptional case, we should ignore each issue directly on Codacy, so that in the following analysis Codacy will no longer report the issues.

Reply:
Done!
I'm unfamiliar with the internals of the ignore logic. It's a bit unclear to me if these issues will resurface in the future. I see that there's also an option to ignore the issue altogether, which I tested in 06569e2da1b09da90bea165f01ac3a71840acf80 and later reverted. It's probably overkill, for now at least.

Reply:
The configuration in 06569e2da1b09da90bea165f01ac3a71840acf80 turns off the markdownlint rule completely while ignoring the issues on Codacy only ignores these particular instances where we want to force using this syntax. :+1:

Reply:
Right. I was just wondering how "stable" these ignore rules are. By "unfamiliar with the internals" I mean that I haven't investigated if we'll still see these errors if we modify the file by adding lines, changing the relevant lines, and on.

Om second thought, I'm assuming that as long as git doesn't flag those lines as modified, they won't be checked again, so we should be safe unless those titles change.

Reply:
> I'm assuming that as long as git doesn't flag those lines as modified, they won't be checked again, so we should be safe unless those titles change.

Yes, that's exactly it! :heavy_check_mark: